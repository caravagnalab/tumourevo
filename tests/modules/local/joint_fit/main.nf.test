/* NOTES:
* Let's test it for input coming from SPN01
* which has stopped only at CTREE.
*/
nextflow_process {
    name "Test Process JOINT_FIT"
    script "modules/local/joint_fit/main.nf"
    process "JOINT_FIT"

    test("Should run without failures") {
        when {
            params {
                // define module specific parameters here. Example:
            }
            process {
                """
                meta = [
                    id : "MSeq_Set06_Set6_42",
                    dataset : "MSeq",
                    patient : "Set06",
                    tumour_sample: "Set6_42",
                    cancer_type: "PANCANCER" ,
                    cna_caller : "sequenza"
                ]
                rds_join = file(params.local_testdata_base_path + '/from_joinCNAqc/MSeq_Set06_multi_cnaqc.rds', checkIfExists: true)
                best_fit =file(params.local_testdata_base_path + '/from_mobsterh/MSeq_Set06_Set6_42_mobsterh_st_best_fit.rds', checkIfExists: true)
                input[0] = tuple(meta, rds_join, best_fit, meta.tumour_sample)
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
