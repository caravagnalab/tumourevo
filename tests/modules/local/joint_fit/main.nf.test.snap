{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "MSeq_Set06_Set6_42",
                            "dataset": "MSeq",
                            "patient": "Set06",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER",
                            "cna_caller": "sequenza"
                        },
                        "MSeq_Set06_Set6_42_all_multi_cnaqc_filtered.rds:md5,8a41b72b75f72c389e35e27e9a8bab96",
                        "Set6_42"
                    ]
                ],
                "rds": [
                    [
                        {
                            "id": "MSeq_Set06_Set6_42",
                            "dataset": "MSeq",
                            "patient": "Set06",
                            "tumour_sample": "Set6_42",
                            "cancer_type": "PANCANCER",
                            "cna_caller": "sequenza"
                        },
                        "MSeq_Set06_Set6_42_all_multi_cnaqc_filtered.rds:md5,8a41b72b75f72c389e35e27e9a8bab96",
                        "Set6_42"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-09-24T15:18:06.17667704"
    }
}